src/private.jl

```diff
@@ -369,6 +370,36 @@ function _pt(
         end
     end
 
+    # Verify if we have a circular reference.
+    ptd = get(io, :__PRETTY_TABLES_DATA__, nothing)
```

Comment:
Do I understand the logic correctly that this records all objects that were passed for printing by `pretty_table`? Then for `DataFrame` it is OK, but what if the default printing of some object does not use `pretty_table` (probably this cannot be helped and one of the tests I proposed below will error, but I want to be sure I understand the logic)?

Things look good in general. Thank you!

Reply:
Thanks!

My approach is:

Every time an object is printed with PrettyTables, I add it in this vector. All cells are rendered using an `IOContext` with this vector. Thus, any object that calls `pretty_table` is recorded.

If the default printing of an object does not use `pretty_table`, I do not know if we can do something about it.

Reply:
> I do not know if we can do something about it.

We cannot. This is what I thought you do. So all is OK.

Reply:
The reason why your code is OK is that immutable values cannot currently be self-refferential in Julia.

Reply:
Yes! I though about it and then I realize that it is just not a problem.